### ISSUE TITLE
Word count of selected text

### ISSUE THREAD

**Issue by stur86** [marktext/marktext#2791]
### Describe your feature request

Just as mentioned in https://github.com/marktext/marktext/issues/2274. It would be useful to have the ability to see the word/character count not just for the whole file, but for the selected text as well. It could be done by showing an additional number in the top left corner (for example in parentheses next to the global count), or somewhere else.

### What problem does this feature solve?

It's useful when for example one is writing text with constrained word count that needs to be only partially copy-pasted somewhere else, when trying to estimate how much one has written in a session, and so on.

**Comment by littlePING-221** [comment]
I agree, Typora has this feature, it has "total/selected" word count in its bottom left corner. It would be great if Marktext also have this feature.

**Comment by kass-kass** [comment]
+1 on this! Very useful when writing down answers to long forms/applications, where each answer specifies its own word count (i.e. bio 100 words, project description 300 words). 

I assume it can also be linked to the setting of the counter: if the counter is set to words, count selection words, if counter is set to characters, count selection characters, etc.

**Comment by plum** [comment]
Agreed, as all students who are writing have a word count limit for their submission and in cosideration of;

**The wrting Process**

- Usually a collection  of notes from reading and reflection is incorporated in a first  draft far exceeds the word count. limit. 
- Not only does a writer needs to check the count before the piece is submitted, but also to _keep an eye on it as they write_ a large peice of say 10,000 words, in order to choose how and where best condense sections they have in a draft.  
- That is achieved  by constantly scanning the document, whilst editing away and moving between sections, keeping an eye on the word count as it decreases.

---

> I could send you a script which can count the words in Vim if you like.

**Comment by Jocs** [comment]
This is implemented now.

The title-bar counter shows the selection count alongside the document count (`document / selection`), and the tooltip does the same for each row — words, characters, paragraphs and all-characters. It follows whichever metric the counter is set to, which is what the follow-up comment here asked for. It works in both the WYSIWYG and source-code views.

Shipped in **v0.20.0-rc.5** (`ba5d71df`) — so it is in the 0.20.0 pre-releases, not in stable 0.19.1.

Closing as completed.
